Libraries/config/Sources/config.cpp

```diff
@@ -0,0 +1,40 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <config/Config.h>
+#include <plist/Format/Any.h>
+#include <plist/Operations/Operations.h>
+
+#include <memory>
+#include <vector>
+
+std::unique_ptr<plist::Object>
+config::readDefaults(libutil::Filesystem const *filesystem) {
+   static std::string defaultConfigPath = "/var/db/xcbuild_defaults.plist";
+
+    std::pair<bool, std::vector<uint8_t>> result = plist::Read(filesystem, defaultConfigPath);
```

Comment:
This could probably just be:

    std::vector<uint8_t>> contents;
    if (!filesystem->read(&contents, defaultConfigPath)) {
         fprintf(stderr, "error: unable to read defaults\n");
         return nullptr;
    }

It seems like the same or less code as with `plist::Read()`, so I was wondering if that was needed at all. It doesn't look like you'd be using the functionality to interpret "-" specially, and `plist::Read()` doesn't do anything else.